### ISSUE TITLE
/etc/default/grub is incompatible with CentOS systems

### ISSUE THREAD

**Issue by ignatenkobrain** [OSInside/kiwi#2139]
## Problem description

Default grub config generated by kiwi has things like `GRUB_TERMINAL="gfxterm"` which on the boot produce errors:

![image](https://user-images.githubusercontent.com/2866862/168478688-cc5016ee-8a0e-4883-83a9-2e3440fad24f.png)

Kiwi: https://github.com/OSInside/kiwi/blob/b56aaed4da4b6ddf8a9028adcaa1f96ad3988845/kiwi/bootloader/config/grub2.py#L632-L742
Anaconda: https://github.com/rhinstaller/anaconda/blob/a1ff62e575294b33d7870072d9befe98278224e2/pyanaconda/modules/storage/bootloader/grub2.py#L253-L287

## Expected behaviour

Kiwi configures grub2 on CentOS/Fedora systems the way it is configured by "Red Hat tooling" so that no errors appear.

## Steps to reproduce the behaviour

Build some minimalistic expandable CentOS Stream 9 image with grub2 as a bootloader and install it on the real hardware :)

## OS and Software information
+ KIWI version: 9.24.34
+ Operating system: CentOS Stream 9 (Fedora 36 on host)
+ OBS version: N/A

**Comment by schaefi** [comment]
Going to create a reproducer image and sending for review @Conan-Kudo / @iammattcoleman who offered their help. Thanks

**Comment by schaefi** [comment]
@Conan-Kudo I did sent the details for the reproduced a while ago but didn't got feedback so far. I assume this one can be closed ?

**Comment by Conan-Kudo** [comment]
No, I don't recall getting the reproducer. It probably fell by the wayside due to crazy IRL things.

**Comment by ignatenkobrain** [comment]
The problem is still there I believe, so pls don't close this without fix :)

**Comment by schaefi** [comment]
ok so I need to redo this as I don't have the image around anymore :/  ... needs to wait until I find the time to look into it again

**Comment by kagehisa** [comment]
Same Problem with the Ubuntu Focal Image build with the official description, kiwi adds 
```
GRUB_TIMEOUT=10
GRUB_GFXMODE=auto
GRUB_TERMINAL="gfxterm"
```
during the create phase grub2-mkconfig step.

```
[ DEBUG   ]: 08:09:12 | "grub2-mkconfig": in paths "/home/manji/KIWI/build/usr/sbin:/home/manji/KIWI/build/usr/bin:/home/manji/KIWI/build/sbin:/home/manji/KIWI/build/bin:/home/manji/KIWI/build/usr/local/bin:/home/manji/KIWI/build/usr/local/sbin" exists: "False" mode match: not checked
[ DEBUG   ]: 08:09:12 | EXEC: [grep -q GRUB_USE_LINUXEFI /home/manji/KIWI/build/usr/sbin/grub-mkconfig]
[ DEBUG   ]: 08:09:12 | "grub2-mkconfig": in paths "/home/manji/KIWI/build/usr/sbin:/home/manji/KIWI/build/usr/bin:/home/manji/KIWI/build/sbin:/home/manji/KIWI/build/bin:/home/manji/KIWI/build/usr/local/bin:/home/manji/KIWI/build/usr/local/sbin" exists: "False" mode match: not checked
[ DEBUG   ]: 08:09:12 | EXEC: [grep -q GRUB_ENABLE_BLSCFG /home/manji/KIWI/build/usr/sbin/grub-mkconfig]
[ INFO    ]: 08:09:12 | Writing grub2 defaults file
[ INFO    ]: 08:09:12 | --> GRUB_GFXMODE:auto
[ INFO    ]: 08:09:12 | --> GRUB_TERMINAL:"gfxterm"
[ INFO    ]: 08:09:12 | --> GRUB_TIMEOUT:10
[ INFO    ]: 08:09:12 | Writing sysconfig bootloader file
[ INFO    ]: 08:09:12 | --> DEFAULT_APPEND:"root=UUID=3cf83610-491e-4b93-bf39-715111580cc2 "
[ INFO    ]: 08:09:12 | --> FAILSAFE_APPEND:"root=UUID=3cf83610-491e-4b93-bf39-715111580cc2  ide=nodma apm=off noresume edd=off nomodeset 3 "
[ INFO    ]: 08:09:12 | --> LOADER_LOCATION:none
[ INFO    ]: 08:09:12 | --> LOADER_TYPE:grub2-efi
[ INFO    ]: 08:09:12 | --> SECURE_BOOT:yes
```

On HyperV this leads to the following error at boot:
![Focal-HyperV](https://user-images.githubusercontent.com/37015428/214509930-f75d981d-9614-41b3-b315-7b24ed3808f1.PNG)

Which is weird because the module `gfxterm` exists in  `/boot/grub/x86_64-efi`. 
Nonetheless it would be nice to have the grub defaults file used "as is" after the prepare phase of a build.
